doc/weakly_supervised.rst

```diff
@@ -225,16 +225,34 @@ Residual
     .. [2] Adapted from https://gist.github.com/kcarnold/5439917
 
 
+.. _sdml:
+
 SDML
 ----
 
 `SDML`: An efficient sparse metric learning in high-dimensional space via 
-double regularization: L1-penalized on the off-diagonal elements of Mahalanobis
-matrix :math:`\mathbf{M}` and the log-determinant divergence between 
+double regularization: an L1-penalization on the off-diagonal elements of the 
+Mahalanobis matrix :math:`\mathbf{M}`, and a log-determinant divergence between 
 :math:`\mathbf{M}` and :math:`\mathbf{M_0}` (set as either :math:`\mathbf{I}` 
 or :math:`\mathbf{\Omega}^{-1}`, where :math:`\mathbf{\Omega}` is the 
 covariance matrix).
 
+The formulated optimization on the semidfinite matrix :math:`M` is convex:
+
+.. math::
+
+    \min_{\mathbf{M}} = \text{tr}((M_0 + \eta XLX^{T})\cdot M) - \log\det M 
+    + \lambda ||M||_{1, off}
+
+where :math:`\mathbf{X}=[x_1, x_2, ..., x_n]`, :math:`\mathbf{L = D − K}` is 
+the Laplacian matrix, :math:`\mathbf{D}` is a diagonal matrix whose diagonal 
+elements are the sums of the row elements of :math:`\mathbf{K}`, 
+:math:`\mathbf{K}` is the incidence matrix to encode the (dis)similarity 
+information as :math:`\mathbf{K}_{ij} = 1` if :math:`(x_i,x_j)\in \mathbf{S}`, 
+:math:`\mathbf{K}_{ij} = -1` if :math:`(x_i,x_j)\in \mathbf{D}`, 
+:math:`||\cdot||_{1, off}` is the off-diagonal L1 norm of :math:`\mathbf{M}`.
```

Comment:
there are a couple of issues here:
- please inverse the order: start with **K**, then **D**, then **L** so that everything needed has been defined when you introduce them
- **D** for dissimilar pairs conflicts with the matrix **D**. I suggest not to introduce new notations and simply say things like  :math:`\mathbf{K}_{ij} = 1` if :math:`(x_i,x_j)` is a similar pair